### PR TITLE
Added random number generation support

### PR DESCRIPTION
**Pull request by ghost**
This adds generic support for the various random number generation mechanisms on all the platforms.  This is a precursor commit to adding a DQL RANDOM() expression.

### CODE REVIEW

File: lib/Doctrine/DBAL/Platforms/AbstractPlatform.php [hunk 1]
```diff
@@ -2661,6 +2661,16 @@ public function getCurrentTimestampSQL()
     }
 
     /**
+     * Returns the SQL specifier for the platform to get a random number
+     *
+     * @return string
+     */
+    public function getRandomFunctionSQL()
+    {
+        return 'RAND()';
```
Comment (deeky666) [inline lib/Doctrine/DBAL/Platforms/AbstractPlatform.php]: I doubt this is SQL-92 standard. Maybe better to throw an exception here in the first place.

File: lib/Doctrine/DBAL/Platforms/PostgreSqlPlatform.php [hunk 2]
```diff
@@ -1054,6 +1054,14 @@ public function getReadLockSQL()
     /**
      * {@inheritDoc}
      */
+    public function getRandomFunctionSQL()
+    {
+        return 'random()';
```
Comment (deeky666) [inline lib/Doctrine/DBAL/Platforms/PostgreSqlPlatform.php]: The documentation states it produces a number like this:

`random value in the range 0.0 <= x < 1.0`

Not really intended, is it?

File: lib/Doctrine/DBAL/Platforms/OraclePlatform.php [hunk 3]
```diff
@@ -481,6 +481,14 @@ public function getDropViewSQL($name)
     }
 
     /**
+     * {@inheritDoc}
+     */
+    public function getRandomFunctionSQL()
+    {
+        return 'DBMS_RANDOM.VALUE';
```
Comment (deeky666) [inline lib/Doctrine/DBAL/Platforms/OraclePlatform.php]: Documentation states it produces a number like this:

`DBMS_RANDOM.VALUE produces numbers in [0,1) with 38 digits of precision.`

Not really inteneded, is it?

**Comment by deeky666** [comment]
You should clarify on your exact purpose. "random number" is somewhat unclear. Seems like most DB vendors support random decimal numbers between `0` and `1` and integers values between their specific integer boundaries...

**Comment by zeroedin-bill** [comment]
I'm not sure we should include this at all. Using DBMS random number features in a query is almost never a good idea.

**Comment by deeky666** [comment]
Closing accodring to https://github.com/doctrine/doctrine2/issues/1412

**Pull request closed without merging**